anaconda_project/plugins.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+# ----------------------------------------------------------------------------
+# Copyright © 2016, Continuum Analytics, Inc. All rights reserved.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+"""Plugins assets."""
+import os
+from os.path import join
+from anaconda_project.project_commands import (_ArgsTransformer, ProjectCommand, _http_specs)
+
+from entrypoints import get_group_named
+
+
+def get_all(entry_points_group='anaconda_project.plugins'):
+    """Return all the plugins registered."""
+    return {name: plugin.load() for name, plugin in sorted(get_group_named(entry_points_group).items())}
+
+
+class ArgsTrasformerTemplate(_ArgsTransformer):
+    """Template class for plugins args trasformers.
+    Plugins args transformers should subclass it and redefine add_class
+    to implement custom arguments mapping.
+    """
+
+    def __init__(self, command):
+        """Construct an ArgTransformer for the given command.
+
+        Args:
+            command (ProjectCommand): command that maps to the ArgsTransformer
+        """
+        # TODO: document specs
+        # TODO: decide the best way to let users define specs for their
+        #       transformers! Most likely specs should go on the command
+        #       itself?
+        super(ArgsTrasformerTemplate, self).__init__(_http_specs)
+        self.command = command
+
+    def add_args(self, results, args):
+        """Overwrite this method to add custom arguments transformation.
+        It should forwarding the arguments that are custom to the
+        specific command served by this trasformer (
+            i.e., '--anaconda-project-host' --> 'host'
+        )
+
+        Inputs:
+            - results [list(tuples)]: list of 2 element tuples the following:
+                           * option (str): name of the option
+                            * values (lst(str)): list of the values passed
+                                        for the option
```

Comment:
I don't actually understand the above docs, though I think they are describing some code I wrote :-) 
I would think the plugin API should be pretty general, like it takes a list of args and returns a new list of args; it shouldn't expose the specific implementation details of the current args transformers.